engine/src/test/java/org/camunda/bpm/engine/test/concurrency/DeleteProcessDefinitionTest.java

```diff
@@ -113,6 +113,25 @@ public void testDeploymentOfProcessDefinitionWithOrphanEventAndPreviousVersion()
     assertThat(runtimeService.createEventSubscriptionQuery().count()).isEqualTo(1);
   }
 
+  @Test
+  public void testDeploymentOfProcessDefinitionWithOrphanConditionalEvent() {
+    // given
+    String resource = "org/camunda/bpm/engine/test/api/repository/processWithConditionalStartEvent.bpmn20.xml";
+    List<ProcessDefinition> processDefinitions = deployProcessDefinitionTwice(resource);
+    assertThat(processDefinitions.size()).isEqualTo(2);
+    deleteProcessDefinitionsSimultaneously(processDefinitions.get(0).getId(), processDefinitions.get(1).getId());
+
+    assertThat(repositoryService.createProcessDefinitionQuery().list()).isEmpty();
+    assertThat(runtimeService.createEventSubscriptionQuery().count()).isEqualTo(1);
```

Comment:
🙃 Those assertions are not really necessary either. A comment would clarify things for readers of the code.
❓ Is it important that the definitions are deleted simultaneously? Do we even test any concurrency here? We might as well put the test into the non-concurrency [DeleteProcessDefinitionTest](https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/test/java/org/camunda/bpm/engine/test/api/repository/DeleteProcessDefinitionTest.java).

```suggestion
    // Delete both process definitions
    deleteProcessDefinitionsSimultaneously(processDefinitions.get(0).getId(), processDefinitions.get(1).getId());
```

Reply:
1. Removed the assertions.
2. It is important to remove them simultaneously because this is a necessary step to reproduce the scenario in which we have an orphan subscription in the database.